README.md

```diff
@@ -179,6 +186,14 @@ more detectable as a proxy, but it will make the Cloak client detect internet in
 data, after which the connection will be closed by Cloak. Cloak will not enforce any timeout on TCP connections after it
 is established.
 
+`LoopbackTcpSendBuffer` is the number of bytes to use for the tcp loopback send buffer. Use a low value like 4096 to reduce upload bufferbloat on client.
```

Comment:
Given that this parameter controls the send buffer from ck-client to e.g. shadowsocks client, shouldn't this be relevant to download instead of upload? And `LoopbackTcpReceiveBuffer` below should affect upload

Reply:
There's cases where cloak is used in tunnel mode not with ss, so user would upload to cloak and then to cloak server directly, not to ss first. This parameter is made for that case.

For usage with ss, it will be ss local receive that needs clamping, in such case buffer between ss client and ck-client will also need clamping (either from ss send side or ck receive side), I have implemented that in ss-rust in the past and it's part it for some time now already.

Reply:
Yes, that's why the ck-client tcp receive needs to be clamped. I put controls for all 4 sides, local send/ local receive / remote send / remote receieve for client and server.

The general principle is all the parts that are on the same machine need to have clamped buffers, and only the ends exposed to the remote internet need to have open buffers, otherwise, bufferbloat will happen.

Reply:
I understand the need for clamping, my issue is with the line
> Use a low value like 4096 to reduce upload bufferbloat on client.

This should describe `LoopbackTcpReceiveBuffer`, not here.

Reply:
It may be better to name these `LocalTcpUploadBuffer`, `LocalTcpDownloadBuffer`, `RemoteTcpUploadBuffer` and `RemoteTcpDownloadBuffer`

Reply:
Yes renaming it makes sense it was very confusing even to me. I will check and update this PR.

Reply:
For client the rename makes sense but for the server not. Because loopback for server point of view means any direction of server that's connected on loopback IP whether it's to the "proxybook" or to the "incoming client"